### PR TITLE
Feat: Queue read connections OPSQLite

### PR DESCRIPTION
**Pull request by mugikhan**
## Work done

- 5 concurrent read-only connections
- Use a queue to manage queries when all connections are busy
- Process queued operations as soon as connections become available

## Testing

Tested locally running multiple queries running at the same time with random delays between the queries.

### CODE REVIEW

File: packages/powersync-op-sqlite/src/db/OPSqliteAdapter.ts [hunk 1]
```diff
@@ -145,36 +147,45 @@ export class OPSQLiteDBAdapter extends BaseObserver<DBAdapterListener> implement
...
-    const sortedConnections = this.readConnections!.map((connection, index) => ({
-      lockKey: `${LockType.READ}-${index}`,
-      connection
-    })).sort((a, b) => {
-      const aBusy = this.locks.isBusy(a.lockKey);
-      const bBusy = this.locks.isBusy(b.lockKey);
-      // Sort by ones which are not busy
-      return aBusy > bBusy ? 1 : 0;
+    return new Promise(async (resolve, reject) => {
+      const execute = async () => {
```
Comment (stevensJourney) [inline packages/powersync-op-sqlite/src/db/OPSqliteAdapter.ts:157]: I think we could remove the `execute` function definition here and just do this inline.

File: packages/powersync-op-sqlite/src/db/OPSqliteAdapter.ts [hunk 2]
```diff
@@ -145,36 +147,45 @@ export class OPSQLiteDBAdapter extends BaseObserver<DBAdapterListener> implement
...
+        // If we have an available connection, use it
+        if (availableConnection) {
+          availableConnection.busy = true;
+          try {
+            resolve(await fn(availableConnection.connection));
+          } catch (error) {
+            reject(error);
+          } finally {
+            // After query execution, process any queued tasks
+            this.processQueue();
```
Comment (stevensJourney) [inline packages/powersync-op-sqlite/src/db/OPSqliteAdapter.ts:170]: Should this set `busy` to false?

File: packages/powersync-op-sqlite/src/db/OPSqliteAdapter.ts (hunk 1 above)
Comment (stevensJourney) [reply packages/powersync-op-sqlite/src/db/OPSqliteAdapter.ts:157]: Sorry, I see this uses recursion

Comment (mugikhan) [reply packages/powersync-op-sqlite/src/db/OPSqliteAdapter.ts:157]: It uses the execute function to push it to the queue. Without it it makes the processQueue repeat what is done in the read lock.

**Review by stevensJourney — APPROVED** [review]

**Pull request merged**